File: server/model/producto.py

from dataclasses import dataclass


@dataclass
class Producto():
    id:int
    nombre:str
    precio:float
    url:str


    def to_dict(self) -> dict:
        return {
            'id': self.id,
            'nombre': self.nombre,
            'precio': self.precio,
            'url': self.url
        }

File: angular/src/model/Producto.ts
export class Producto{
    id: number;
    nombre: string;
    precio: number;
    url:string;

    constructor(id: number, nombre: string, precio: number,url:string){
        this.id = id;
        this.nombre = nombre;
        this.precio = precio;
        this.url = url;

    }
}
